src/components/AccordionSection/AccordionSection.test.tsx

```diff
@@ -50,12 +49,12 @@ describe("AccordionSection ", () => {
     );
     wrapper.find(".p-accordion__tab").at(0).simulate("click");
 
-    expect(onTitleClick.mock.calls[0]).toEqual([true, MOCK_UUID]);
+    expect(onTitleClick.mock.calls[0]).toEqual([true, "mock-nanoid-1"]);
```

Comment:
How do we know when writing tests what the id is gonna be? They reset to 0 every time? We need to know how many times nanoid is called in the components?

Also it seems that now when writing tests we need to know how this mocked version of nanoid works, that it returns ids starting with `mock-nanoid-`, etc. 

Not sure how problematic it may be in long term, but seems dev need to know much more about out specific setup of mocks.

Reply:
For each test (each `it()`) the id starts at `...-1`. Without generating unique ids like this every element gets exactly the same id which causes clashes, which means that "<label for=.." aria labels etc. are associated with the wrong item (I guess the first item with an id).

If we don't mock the id at all then snapshots will receive new ids for every run, meaning that the snapshots won't pass.

This mock doesn't prevent a dev from using jest.mock in a single test to override the ids if they want to explicitly set an id (just like they could before), but also means that for the majority of cases it will just work.

Do you think that seems reasonable? Is there another approach you can think of?

Reply:
@huwshimi I understand why we need it, it's just that to write a test with id now dev needs to know how the nanoid mock is setup and how the id name will be generated. And looking at the tests in 90% of cases we only need one id.

Maybe we can keep `MOCK_UUID` const with `mock-nanoid-1` value, so for all those basic cases when 1 id is needed it can still be used? But I guess this will make the cases when you need more ids more confusing…

I don't have any better idea at the moment.

Reply:
I've changed the Accordion tests to explicitly mock nanoid for the value we expect, do you think that makes it easier to understand?

Reply:
Yes, I think so. It's in the same test file, so it's easier to understand where it comes from. 

I know that the nanoid mock with incremental ids does kind of the same thing "automatically" so you don't have to copy it over and over between test files, but it also hides it from developer who writes or reads the tests. Having it mocked in the test file is more verbose (in a good way I think).

Reply:
Great, thanks @bartaz let's try and use that pattern moving forward then.